### DISCUSSION TITLE
CoreFreq version 1.92.4 is available

### DISCUSSION THREAD

**Discussion by cyring** [cyring/CoreFreq#345] [Show and tell]
![2022-06-11-231330_644x704_scrot](https://user-images.githubusercontent.com/11563789/173205051-62fadd65-6c1c-436f-b3c4-985989240130.png)

@inglor Hello

New version [`1.91.1`](https://github.com/cyring/CoreFreq/releases/tag/1.91.1) is available 

Thank you for your package.

**Comment by cyring** [comment]
* Tips: Do your guys know _CoreFreq_ can scale up to 1024 CPU !

```sh
make CORE_COUNT=1024
```

Here is a KVM accelerated simulation of 254 monitored CPU

```sh
qemu-system-x86_64 -enable-kvm -cpu max,hv-runtime -smp 254 -m 22G -cdrom archlinux-corefreq-dev.iso
```

![2022-06-11-234138_1280x825_scrot](https://user-images.githubusercontent.com/11563789/173205878-6affac2b-c0a7-4e43-93a2-4ee9ffcc8e38.png)

See the Help windows for various scrolling shortcut keys  

![2022-06-11-234211_1280x825_scrot](https://user-images.githubusercontent.com/11563789/173205884-89097323-60e6-41c9-9db8-af3eb57ba52f.png)

* Tips: Do your guys know some features are build time option ?

```sh
make help
```

For example, the AMD/Zen UMC relative frequencies are available from the `Package` view
```sh
make ARCH_PMC=UMC

corefreq-cli -t package
```
![2022-06-11-235651_644x704_scrot](https://user-images.githubusercontent.com/11563789/173206230-290f59ab-3947-4755-854b-d09f22db3b5b.png)

![2022-06-11-235855_644x704_scrot](https://user-images.githubusercontent.com/11563789/173206262-8742c62b-0bce-4b54-9d49-81194b9aa5b8.png)

And you can also decide which area will be shown !

Here, just the bar charts:

```sh
make NO_HEADER=1 NO_FOOTER=1 NO_LOWER=1 clean corefreq-cli
```

![2022-06-12-000501_644x466_scrot](https://user-images.githubusercontent.com/11563789/173206354-cf530647-1eba-4310-94df-99979e5c82dc.png)

Enjoy _CoreFreq_

CyrIng

**Comment by cyring** [comment]
@inglor @ich777 Hello

[`1.91.2`](https://github.com/cyring/CoreFreq/releases/tag/1.91.2) is an update of the `master` version, including fixes for latest issues:
* Safely collect Linux memory information (Spinlock issue #347)
* Count all processes
* CentOS Kernel 3.10 api compatibility
* Startup Frequency view: display the AMD/Zen UMC channels frequencies instead of the unavailable package C-States. Press `%` to toggle to this UMC footer. 
  **EDIT**: only tested on my Matisse and Lucienne, no other Users report, so it is still a WiP feature, need to build with directive `ARCH_PMC=UMC`

**Reply by ich777** [reply]
@cyring thank you for the update!
Should I compile the AMD version for Unraid with UMC?
My build toolchain should pick up the new version in about an hour and automatically start the build.

**Reply by cyring** [reply]
Building the AMD version `make ARCH_PMC=UMC` does not impact runtime for Intel.
So you can indeed propose that way to the Unraid users.

**Reply by ich777** [reply]
Changed the build script and it has already compiled the AMD version from CoreFreq `v1.91.2` with UMC

**Reply by cyring** [reply]
Did you try it?

**Reply by ich777** [reply]
No I actually can‘t because I don‘t have any AMD system on hand.

**Reply by cyring** [reply]
Concerning Intel processors, this version has the frozen Spinlock  fix for this Unraid [case](https://forums.unraid.net/topic/124892-unraid-becomes-unresponsive/#comment-1138303)

**Reply by cyring** [reply]
> Changed the build script and it has already compiled the AMD version from CoreFreq `v1.91.2` with UMC

The `ARCH_PMC=UMC` is made for AMD only. 
Using it on Intel will mess with the UI/Package C-states view:

![2022-06-21-004258_642x427_scrot](https://user-images.githubusercontent.com/11563789/174685695-431c69ef-6ea5-4d58-b824-9b9e3ffd5fff.png)

**Comment by cyring** [comment]
_CoreFreq_ refreshed in `1.91.3`

Example:
* Intel processor HWP and OC capable
```
insmod corefreqk.ko Ratio_HWP=12,47,47 Ratio_Boost=-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,46,46,47 Ratio_PPC=47 HWP_Enable=1
```
* AMD with P-state ratios unlocked
```
insmod corefreqk.ko Ratio_Boost=-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,10,24 TurboBoost_Enable=0,1
```

**Reply by cyring** [reply]
Latest commit a485cecf002c950c97955e81d20d87059115f7f2 merged into `master` is adding driver parameter `Ratio_PPC` which is known as Performance 
Target ratio in the Client. 



----

FYI, pausing _CoreFreq_ developments for a while.

**Reply by ich777** [reply]
> FYI, pausing _CoreFreq_ developments for a while.

Well deserved, enjoy your time off... :)

**Comment by cyring** [comment]
@inglor @ich777 Hello

A SMBIOS strings fix is available in version [`1.91.4`](https://github.com/cyring/CoreFreq/releases/tag/1.91.4)

Thank you

**Comment by ich777** [comment]
@cyring my build machine already notified my and built it already for Unraid 6.10.3 and I will trigger the build manually for 6.11.0-rc2 now. :)

**Comment by cyring** [comment]
@inglor and for anyone of packaging purpose

Version [`1.91.5`](https://github.com/cyring/CoreFreq/releases/tag/1.91.5) is released into master branch

1. AMD Threadripper 3960X UMC improvements
2. Intel Alder lake IMC channels counting fix
3. AMD Ryzen Cezanne UMC decoding fix
4. SMBIOS DIMM strings improvements

**Comment by cyring** [comment]
@inglor Hello,

Version [`1.91.6`](https://github.com/cyring/CoreFreq/releases/tag/1.91.6) is available for packaging

**Reply by ich777** [reply]
I've triggered the build manually, usually my tool-chain checks every 2 hours for updates. :)

**Reply by ich777** [reply]
Oh, BTW does this version have already the changes from `develop` in it since I have to build it for @SimonFair?

**Reply by cyring** [reply]
> Oh, BTW does this version have already the changes from `develop` in it since I have to build it for @SimonFair?

Yes. Only debug traces have been removed.

**Reply by SimonFair** [reply]
Hi @ich777 should I be able to install via CA?

**Reply by ich777** [reply]
@SimonFair sorry for the late reply. Compiled it and will send it in a few minutes on the Unraid forums to you.

**Comment by cyring** [comment]
@inglor 

CoreFreq version [`1.91.7`](https://github.com/cyring/CoreFreq/releases/tag/1.91.7) is available with a fix for Intel/Alder Lake with Pcores only, like i5-12500

**Comment by cyring** [comment]
@inglor 
Hello version `1.92.0` is available into master branch
![2022-09-12-145948_432x153_scrot](https://user-images.githubusercontent.com/11563789/189660158-45718c58-07e4-4f06-9009-613a762dc417.png)

**Comment by cyring** [comment]
![2022-10-21-193053_644x1012_scrot](https://user-images.githubusercontent.com/11563789/197254689-cd233150-37a8-47b8-9d0f-33b8e8544238.png)

**Reply by ich777** [reply]
Already compiled. 😉

**Reply by cyring** [reply]
Thanks a lot

Intel Mitigation mechanisms have been improved, you should discover various features.

AMD Zen3+ is supported.
IMC timings for Rembrandt appears fine but not geometry.

From SandyBridge, up to Broadwell, IMC and clocking have been improved also.

**Comment by cyring** [comment]
_CoreFreq_ [`1.92.3`](https://github.com/cyring/CoreFreq/releases/tag/1.92.3) allows building against Kernel versions `6.0.3` and `6.1.0-rc1`

![2022-10-23-003624_644x1012_scrot](https://user-images.githubusercontent.com/11563789/197364826-abf9decf-f3b1-456a-8649-7d67062b099d.png)

**Comment by cyring** [comment]
@ich777 @inglor Hello,

Fyi version [1.92.4](https://github.com/cyring/CoreFreq/releases/tag/1.92.4) is a hotfix
* to let the build against kernel `<` `4.18` 
* to prevent a driver freeze with `Skylake/X` and derivatives of CPUID `6.55`
